src/api/githubService.js

```diff
@@ -0,0 +1,284 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import axios from "axios";
+import moment from "moment";
+
+const functionalTestsWorkflowId = 5937682;
+
+export const getAllWorkflows = async () => {
+  const api = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/actions/runs?branch=main`;
+
+  let wfRunsInfo = [];
+  let workflows = {};
+
+  if (await _isWithinExpiryTime("awf_expireTime"))
+    return await _getDataFromLocal("all_workflows");
+
+  const {
+    data: { workflow_runs },
+  } = await axios.get(api);
+
+  for (let workflow_run of workflow_runs) {
+    if (workflows[workflow_run.name] === undefined) {
+      workflows[workflow_run.name] = workflow_run.id;
+      wfRunsInfo.push(workflow_run);
+    }
+  }
+
+  window.localStorage.setItem(
+    "awf_expireTime",
+    moment().add(45, "m").valueOf()
+  );
+  window.localStorage.setItem("all_workflows", JSON.stringify(wfRunsInfo));
+  return wfRunsInfo;
+};
+
+export const getAllRunsForWorkflow = async (workflow_id) => {
+  const api = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/actions/workflows/${workflow_id}/runs?branch=main&per_page=100`;
+
+  if (await _isWithinExpiryTime(`awfs_expireTime_${workflow_id}`))
+    return await _getDataFromLocal(`awfs_workflows_${workflow_id}`);
+
+  const {
+    data: { workflow_runs },
+  } = await axios.get(api);
+
+  window.localStorage.setItem(
+    `awfs_expireTime_${workflow_id}`,
+    moment().add(45, "m").valueOf()
+  );
+  window.localStorage.setItem(
+    `awfs_workflows_${workflow_id}`,
+    JSON.stringify(workflow_runs)
+  );
+  return workflow_runs;
+};
+
+export const getFullTestWorkflowReport = async () => {
+  let return_data = {
+    fetched: moment().format().valueOf(),
+    workflow_runs_data: [],
+  };
+
+  if (await _isWithinExpiryTime("expireTime"))
+    return await _getDataFromLocal("workflows");
+
+  const apiData = await _getSingleWorkflow(functionalTestsWorkflowId);
+  if (await _checkForUpdate("expireTime", "workflows", apiData))
+    return await _getDataFromLocal("workflows");
+
+  try {
+    // get latest workflow runs for main branch
+    const workflows = await _getSingleWorkflow(functionalTestsWorkflowId);
+
+    // build object for workflow runs and check runs (functional test matrix run) for each workflow run
+    for (let i = 0; i < workflows.length; i++) {
+      let workflow = {
+        workflow_run: {},
+        test_runs: [],
+        passed_count: 0,
+        failed_count: 0,
+        flaked_count: 0,
+      };
+
+      workflow.workflow_run = workflows[i];
+      _delay(Math.floor(Math.random() * 1000) + 200);
```

Comment:
this needs to go away otherwise the loading of the page will take  seconds and seconds...